[inductor] Fix "RuntimeError: Tried to erase Node permute but it still had 3 users in the graph"

jiawenliu64 opened the pull request:
Summary: Fix "RuntimeError: Tried to erase Node permute but it still had 3 users in the graph" to unblock internal models

Differential Revision: D42213859



cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

ngimel commented:
Can you please add a testcase based on @anijain2305 minifier? You can patch `_inductor.config` in the test to enable permute fusion.

jiawenliu64 commented:
Synced with @anijain2305: the issue is addressed with tests in minifier, and add tests based on minifier

jianyuh reviewed (approved):
Rebase to check the OSS test failure?

jiawenliu64 commented:
It seems all test failures are due to the sev (#91332). Will check when the sev is solved

Commit:
[inductor] Fix "RuntimeError: Tried to erase Node permute but it still had 3 users in the graph" (#91327)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/91327

Fix "RuntimeError: Tried to erase Node permute but it still had 3 users in the graph" to unblock internal models

Reviewed By: anijain2305, jianyuh, ngimel

Differential Revision: D42213859

fbshipit-source-id: ae02b2c75cf67e275fb5dd841a238847904a53d9

jiawenliu64 commented:
@pytorchbot merge

jiawenliu64 commented:
@pytorchbot merge -f "failure unrelated"
